test/initialization-test/run_test.sh

```diff
@@ -57,8 +61,11 @@ fi
 GITHUB_REPO=kubeflow/pipelines
 BASE_DIR=/go/src/github.com/${GITHUB_REPO}
 JUNIT_TEST_RESULT=junit_InitializationTestOutput.xml
-TEST_DIR=backend/test/initialization
-
+if [ -n "$TEST_V2_API" ]; then
+  TEST_DIR=backend/test/v2/initialization
```

Comment:
In that way, `go test ./...` on v1 test folder would also run v2 tests because it's a subdirectory. 
Technically subdirectories can still be filtered out, but I don't feel like overcomplicating the command, which could be error-prone.